scintillometry/io/psrfits/hdu.py

```diff
@@ -93,22 +143,53 @@ def frequency(self):
         # and c_freq is the frequency of channel n_nchan / 2.  We use
         # (n_chan + 1) // 2 to ensure this makes sense for n_chan = 1
         # and is consistent with the document at least for even n_chan.
+
         freq = c_freq + (np.arange(1, n_chan + 1) -
                          ((n_chan + 1) // 2)) * chan_bw
         return u.Quantity(freq, u.MHz, copy=False)
 
+    @frequency.setter
+    def frequency(self, freq):
+        """Frequency setter. The input should be the frequency array."""
+        freq = freq.to_value(u.MHz)
+        n_chan = len(freq)
+        # add the channel 0
+        # we assuem the frequency resoluation is the same across the band.
+        freq_pad = np.insert(freq, 0, 2 * freq[0] - freq[1])
+        # Are we doing the right thing? Maybe we should check with psrfits.
```

Comment:
Yes, would be good...

Reply:
Follow the assumption above, I think we are doing the right thing.

Reply:
Then, change the comment to say that we're following the retrieval above. Obviously, this will need a test when we start storing real data.